src/components/Users/UserResetPassword.tsx

```diff
@@ -1,207 +1,222 @@
+      })
+      .regex(/[A-Z]/, {
+        message: "Password must contain at least one uppercase letter",
+      }),
+    new_password_2: z
+      .string()
+      .min(1, { message: "Please confirm your new password" }),
+  })
+  .refine((values) => values.new_password_1 === values.new_password_2, {
+    message: "New password and confirm password must be the same.",
+    path: ["new_password_2"],
+  });
 
 export default function UserResetPassword({
   userData,
 }: {
   userData: UserBase;
 }) {
   const { t } = useTranslation();
-  const [isSubmitting, setisSubmitting] = useState(false);
   const [isEditing, setIsEditing] = useState(false);
-
-  const initForm: PasswordForm = {
-    username: userData.username,
-    old_password: "",
-    new_password_1: "",
-    new_password_2: "",
-  };
-
-  const validateNewPassword = (password: string) => {
-    if (
-      password.length < 8 ||
-      !/\d/.test(password) ||
-      password === password.toUpperCase() ||
-      password === password.toLowerCase()
-    ) {
-      return false;
-    }
-    return true;
-  };
-
-  const validateForm = (formData: PasswordForm) => {
-    const errors: Partial<Record<keyof PasswordForm, string>> = {};
-
-    if (!formData.old_password) {
-      errors.old_password = t("please_enter_current_password");
-    }
-
-    if (!formData.new_password_1) {
-      errors.new_password_1 = t("please_enter_new_password");
-    } else if (!validateNewPassword(formData.new_password_1)) {
-      errors.new_password_1 = t("new_password_validation");
-    }
-
-    if (!formData.new_password_2) {
-      errors.new_password_2 = t("please_confirm_password");
-    } else if (formData.new_password_1 !== formData.new_password_2) {
-      errors.new_password_2 = t("password_mismatch");
-    }
-
-    if (formData.new_password_1 === formData.old_password) {
-      errors.new_password_1 = t("new_password_same_as_old");
-    }
-
-    return errors;
-  };
-
-  const handleSubmit = async (formData: PasswordForm) => {
-    setisSubmitting(true);
+  const [isPasswordFieldFocused, setIsPasswordFieldFocused] = useState(false);
+
+  const resetPasswordForm = useForm({
+    resolver: zodResolver(PasswordSchema),
+    defaultValues: {
+      old_password: "",
+      new_password_1: "",
+      new_password_2: "",
+    },
+  });
+  const { mutate: resetUserPasswordMutate, isPending } = useMutation({
+    mutationFn: async (formData: UpdatePasswordForm) => {
+      const response = await mutate(routes.updatePassword, { silent: true })(
+        formData,
+      );
+      if ("errors" in response) {
+        throw response;
+      }
+      return response;
+    },
+    onSuccess: (data: any) => {
+      Notification.Success({ msg: data?.message as string });
+      resetPasswordForm.reset();
+    },
+    onError: (error: any) => {
+      const errorMessage =
+        error?.response?.data?.message || t("password_update_error");
+      Notification.Error({ msg: errorMessage });
+    },
+  });
```

Comment:
1. follow docs. do not `() => mutate(...)(...)` just do `mutate`
2. `mutate` throws errors for you. you don't need to do it manually.
3. errors notifcations are dispatched by our global error handler. you don't need to do it.


@coderabbitai , maybe update the request's README to include this, since I'm done repeating this same thing on all PRs... :)